packages/remix/src/ssr/types.ts

```diff
@@ -0,0 +1,30 @@
+import { Session, User } from '@clerk/backend-core/src';
+import { ServerSideAuth } from '@clerk/types';
+import { LoaderFunction } from 'remix';
+
+export type InferRootLoaderData<U = any> = {
```

Comment:
Perhaps call this `WithClerkState<>`?

We don't have too much prior-art here, but this is similar:
https://docs.clerk.dev/reference/clerk-react/useuser-hook#typescript-support

Reply:
Hm, not sure about `WithUserProp<MyType>`. I think it means that we're *enhancing* `MyType` with our user obj. 

My intention with `InferRootLoaderData` however, is exactly the opposite: let the user type `data` while obscuring the internals of `clerkState` as much as possible

```tsx
const { data, clerkState } = useLoaderData<InferLoaderData<number>>();
```
 
But I see why the name `InferRootLoaderData` is a bit problematic. Happy to go with `WithUserProp` if we can't come up with something much better. What do you think?

cc: @colinclerk , @SokratisVidros , @igneel64

Reply:
-> WithUserProp

Reply:
WithClerkState is more intuitive to me personally - I don't think it's imperative we obscure this detail of how our integration works

Reply:
+1 for WithClerkState